src/erlcloud_ddb2.erl

```diff
@@ -838,6 +834,65 @@ conditional_op_opt() ->
 expected_opt() ->
     {expected, <<"Expected">>, fun dynamize_expected/1}.
 
+-spec filter_expression_opt() -> opt_table_entry().
+
+filter_expression_opt() ->
+    {filter_expression, <<"FilterExpression">>, fun dynamize_expression/1}.
+
+% This matches the Java API, which asks the user to write their own expressions.
+
+-spec dynamize_expression(expression()) -> binary().
+dynamize_expression(Expression) when is_binary(Expression) ->
+    Expression;
+dynamize_expression(Expression) when is_list(Expression) ->
+    list_to_binary(Expression);
+
+% Or, some convenience functions for assembling expressions using lists of tuples.
+
+dynamize_expression({A, also, B}) ->
+    AA = dynamize_expression(A),
+    BB = dynamize_expression(B),
+    <<"(", AA/binary, ") AND (", BB/binary, ")">>;
+dynamize_expression({{A, B}, eq}) ->
+    <<A/binary, " = ", B/binary>>;
+dynamize_expression({{A, B}, ne}) ->
+    <<A/binary, " <> ", B/binary>>;
+dynamize_expression({{A, B}, lt}) ->
+    <<A/binary, " < ", B/binary>>;
+dynamize_expression({{A, B}, le}) ->
+    <<A/binary, " <= ", B/binary>>;
+dynamize_expression({{A, B}, gt}) ->
+    <<A/binary, " > ", B/binary>>;
+dynamize_expression({{A, B}, ge}) ->
+    <<A/binary, " >= ", B/binary>>;
+dynamize_expression({{A, {Low, High}}, between}) ->
+    <<A/binary, " BETWEEN ", Low/binary, " AND ", High/binary>>;
+dynamize_expression({{A, B}, in}) when is_binary(B) ->
+    <<A/binary, " IN ", B/binary>>;
+dynamize_expression({{A, B}, in}) when is_list(B) ->
+    % Convert everything to binaries.
+
+    InList = [to_binary(X) || X <- B],
+
+    % Join the list of binaries with commas.
+
+    Join = fun(Elem, Acc) when Acc =:= <<"">> ->
+                Elem;
+              (Elem, Acc) ->
+                <<Acc/binary, ",", Elem/binary>> end,
+
+    In = lists:foldl(Join, <<>>, InList),
+
+    <<A/binary, " IN (", In/binary, ")">>;
+dynamize_expression({attribute_exists, Path}) ->
+    <<"attribute_exists(", Path/binary, ")">>;
+dynamize_expression({attribute_not_exists, Path}) ->
+    <<"attribute_not_exists(", Path/binary, ")">>;
+dynamize_expression({begins_with, Path, Operand}) ->
+    <<"begins_with(", Path/binary, ",", Operand/binary, ")">>;
+dynamize_expression({contains, Path, Operand}) ->
+    <<"contains(", Path/binary, ",", Operand/binary, ")">>.
+
```

Comment:
As I commented in #209 ([comment link](https://github.com/gleber/erlcloud/issues/209#issuecomment-142305659)) and #215 ([comment link](https://github.com/gleber/erlcloud/pull/215#issuecomment-145734465)), I'd prefer not to add these complexities to our users.
Sorry but I'm a little busy these days that I couldn't have made some use cases analyses yet. But if I cannot convince you without them, I'll make some later.